fmpz/multi_CRT_ui.c

```diff
@@ -13,126 +14,96 @@
 #include <gmp.h>
 #include "flint.h"
 #include "ulong_extras.h"
+#include "mpn_extras.h"
 #include "fmpz.h"
 #include "nmod_vec.h"
 
+#define MAC(h, l, a, b)                 \
```

Comment:
Is this a candidate for longlong.h? It looks like it could be useful.

Reply:
longlong.h is already such a mes. I would prefer to have this right here next the code that uses it.

Reply:
Honestly, this really seems like a useful function. If you think longlong.h is a mess, you should see the file it was derived from. :-)

Surely this will just be implemented over and over if it doesn't go in longlong.h.